src/struveH.jl

```diff
@@ -0,0 +1,73 @@
+
+struveH(v::Real, x::Real) = _struveH(float(v), float(x))
+struveH(v, x) = _H_integral(v, x)
+
+_struveH(v, x::AbstractFloat) = _H_integral(v, x)
+_struveH(v::Float32, x::Float32) = Float32(_struveH(Float64(v), Float64(x)))
```

Comment:
I feel like this sort of convenience implementation kind of misleads users about the capabilities of the package. It's probably better to just throw an MethodError and admit there isnt a 32-bit / 16-bit implementation yet

Reply:
I'm happy with whatever you think is best. I have just updated `struveH` and `struveL` to have explicit `Float32` routines. These routines are not prone to cancellation in the sums so they can be used pretty indistinguishably. We can even add higher order (`Float128`) methods as well based on this.

Though, it is the standard to cast lower floating point numbers to higher and do the calculation in higher precision (https://github.com/JuliaMath/SpecialFunctions.jl/blob/36c547b4a270b6089b1baf7bec05707e9fb8c7f9/src/beta_inc.jl#L16). Especially for `Float16` as they are just really for storage and not really dedicated hardware for this (it will be faster to do the calculation in `Float32` than do it in `Float16`. 

For `struveK` and `struveH` we need a bit more care as they are prone to cancellation. I will add dedicated Float32 routines for this, however they should also cast the Float32 number to Float64 and do the calculation so those expansions are valid in larger regions. I'm open to whatever you think on this but these routines probably will benefit from doing the calculations all in `Float64` to a desired tolerance.